lib/PrometheusPlugin.ts

```diff
@@ -23,81 +23,157 @@
+       * @default [0.001,0.01,0.1,1,2,5]
+       */
+      gcDurationBuckets?: number[];
+
+      /**
+       * String to prefix default metrics with
+       * @default 'kuzzle_'
+       */
+      prefix?: string;
+    }
+  };
+}
 
 /**
  * @class PrometheusPlugin
  *
- * @property {KuzzlePluginContext}     context        Kuzzle plugin context
- * @property {Object.<string, string>} hooks          Kuzzle plugin hooks
- * @property {Routes}                  routes         Kuzzle plugin routes
- * @property {Controllers}             controllers    Kuzzle plugin controllers
- * @property {Prometheus.Registry}     registry       Prometheus Registry
- * @property {Object.<string, Object>} kuzzleMetrics  Prometheus metrics dedicated for Kuzzle
- * @property {String}                  key            Key used when pushing Prometheus Registry to Redis
- * @property {String}                  hostname       Hostname
- * @property {String}                  MAC            Host MAC address
- * @property {String}                  IP             Host IPv4 adress
+ * @property {PluginConfiguration}     config         Plugin configuration
+ * @property {MetricService}           metricService  Metric service
  *
  * @externs
  */
 export class PrometheusPlugin extends Plugin {
-  private defaultConfig: JSONObject;
-  private prometheusController : PrometheusController;
+  /**
+   * Plugin configuration validation
+   */
+  public config: PrometheusPluginConfiguration;
 
-   /**
-   * PrometheusPlugin constructor
-   *
-   * @returns {PrometheusPlugin}
+  /**
+   * Service to manage, update and format Prometheus metrics
    */
-  constructor () {
+  private metricService: MetricService;
+
+  constructor() {
     super({
-      kuzzleVersion: '>=2.11.0 <3'
+      kuzzleVersion: '>=2.16.0 <3'
     });
 
-    this.defaultConfig = {
-      collectSystemMetrics: true,
-      systemMetricsInterval: 5000,
-      labels: {
-        common: [
-          'nodeHost',
-          'nodeMAC',
-          'nodeIP'
-        ],
-        kuzzle: [
-          'controller',
-          'action',
-          'event',
-          'protocol',
-          'status'
-        ]
+    /**
+     * Default plugin configuration
+     */
+    this.config = {
+      metrics: {
+        default: {
+          enabled: true,
+          prefix: 'kuzzle_',
+          eventLoopMonitoringPrecision: 10,
+          gcDurationBuckets: [0.001, 0.01, 0.1, 1, 2, 5],
+        },
+        core: {
+          monitorRequestDuration: true,
+          prefix: 'kuzzle_',
+        }
       }
     };
   }
 
-  async init (config: JSONObject, context: PluginContext) {
-    this.config = { ...this.defaultConfig, ...config };
+  /**
+   * Plugin initialization
+   * @param {PrometheusPluginConfiguration} config  - Plugin configuration
+   * @param {PluginContext}       context - Kuzzle plugin context
+   */
+  async init(config: PrometheusPluginConfiguration, context: PluginContext) {
```

Comment:
Nitpicking: coding standard
```suggestion
  async init (config: PrometheusPluginConfiguration, context: PluginContext) {
```